### DISCUSSION TITLE
Gem5 full system simulation for RISCV

### DISCUSSION THREAD

**Discussion by kislay536** [gem5/gem5#1473] [Q&A]
Hi,
I was trying to run a custom .riscv workload in gem5 FS mode. I have two options- 
1. with General FS mode with OS-
    I am using this script "build/RISCV/gem5.opt configs/example/riscv/fs_linux.py --kernel=$uboot/uboot.elf --disk-image=$ubuntu_img/ubuntu.img --cpu-type=TimingSimpleCPU --num-cpus=1 --caches --l2cache --l1i_size=32kB --l1d_size=32kB --l2_size=1MB --l2_assoc=2 --mem-type=DDR4_2400_4x16 --mem-ranks=4 --mem-size=4GB --sys-clock=1600MHz", So wanted to ask that where am I supposed to run that binary and how? I have connected to this terminal using the port 3456.
2.  with baremetal
 I don't know how exactly to use this. I tried this "build/RISCV/gem5.opt configs/example/riscv/fs_linux.py --bare-metal --cpu-type=TimingSimpleCPU --num-cpus=1 --caches --l2cache --l1i_size=32kB --l1d_size=32kB --l2_size=1MB --l2_assoc=2 --mem-type=DDR4_2400_4x16 --mem-ranks=4 --mem-size=4GB --sys-clock=1600MHz --kernel=$boom/condBranchMispred.riscv" but I don't know how to see the output.

Please help me with this.

**Comment by ShefaaAbed** [comment]
Hello, I have the same problem now! Did you know how to run your script in FS mode?

**Reply by kislay536** [reply]
Hi,
I don't know how to run this in FS mode. 
Instead, I used "./build/RISCV/gem5.opt configs/example/riscv/fs_linux.py --kernel {path_to_executable} --bare --cpu-type=O3CPU --caches --l2cache" for baremetal simulation, and this was sufficient for my work.

**Reply by ShefaaAbed** [reply]
My problem is that I want to run a python code, the code is just print("hello world"), but still I do not know how to run this python script in FS mode! 
I tried to move the file to the disk_img but still have the same problem. Did you try to run a python code script? or any language except C or C++?

**Reply by kislay536** [reply]
No, I haven't tried any other language and I have used only SE mode since, I was not able to make the FS simulation run
